### ISSUE TITLE
Adding BatchNorm2d reduces validation accuracy

### ISSUE THREAD

**Issue by antimora** [burn-rs/burn#235]
**Describe the bug**

Adding BatchNorm2d  reduces validation accuracy significantly. With each epoch the validate accuracy drops.

**To Reproduce**
 1. Add `BatchNorm2d` layer to `ConvBlock` of `examples/mnist/src/model.rs`.
   ```rust

#[derive(Module, Debug)]
pub struct ConvBlock<B: Backend> {
    conv: Param<nn::conv::Conv2d<B>>,
    norm: Param<nn::BatchNorm2d<B>>,
    activation: nn::GELU,
}

impl<B: Backend> ConvBlock<B> {
    pub fn new(channels: [usize; 2], kernel_size: [usize; 2]) -> Self {
        let conv = nn::conv::Conv2d::new(
            &nn::conv::Conv2dConfig::new(channels, kernel_size).with_bias(false),
        );

        let norm = nn::BatchNorm2d::new(&nn::BatchNorm2dConfig::new(channels[1]));

        Self {
            conv: Param::from(conv),
            norm: Param::from(norm),
            activation: nn::GELU::new(),
        }
    }

    pub fn forward(&self, input: Tensor<B, 4>) -> Tensor<B, 4> {
        let x = self.conv.forward(input);
        let x = self.norm.forward(x);
        self.activation.forward(x)
    }
}

```

2. Run the training `cargo run --example mnist --features tch-gpu --release`

**Expected behavior**
BatchNorm2d should not be making validation accuracy worse.

**Screenshots**

Without BatchNorm2d layers.
```
[Metrics]
  - Train Accuracy: epoch 97.86 % - batch 100.00 %
  - Train Loss: epoch 0.0659 - batch 0.0243
  - Valid Accuracy: epoch 98.46 % - batch 96.67 %
  - Valid Loss: epoch 0.0516 - batch 0.0786
```

With BatchNorm2d layers.
```
[Metrics]
  - Train Accuracy: epoch 97.92 % - batch 98.33 %
  - Train Loss: epoch 0.0701 - batch 0.0270
  - Valid Loss: epoch 1.9889 - batch 1.9347
  - Valid Accuracy: epoch 49.09 % - batch 62.00 %
```

**Additional Information**
I am attaching the training information from `/tmp/burn-example-mnist` directory before the change and after.

 * [burn-example-mnist-batchnorm.zip](https://github.com/burn-rs/burn/files/10983438/burn-example-mnist-batchnorm.zip)
* [burn-example-mnist-no-batchnorm.zip](https://github.com/burn-rs/burn/files/10983439/burn-example-mnist-no-batchnorm.zip)

**Comment by antimora** [comment]
Linking a fix here: https://github.com/burn-rs/burn/pull/238
